src/ontodia/viewUtils/connectionsMenu.tsx

```diff
@@ -401,11 +472,35 @@ interface ConnectionsListProps {
 
     onExpandLink?: (link: FatLinkType, direction?: 'in' | 'out') => void;
     onMoveToFilter?: (link: FatLinkType, direction?: 'in' | 'out') => void;
+
+    propertySuggestionCall?: PropertySuggestionCall;
+    sortMode: SortMode;
 }
 
-class ConnectionsList extends React.Component<ConnectionsListProps, {}> {
+class ConnectionsList extends React.Component<ConnectionsListProps, { scores: Dictionary<PropertyScore> }> {
     constructor (props: ConnectionsListProps) {
         super(props);
+        this.state = { scores: {} };
+        this.updateScores(props);
+    }
+
+    componentWillReceiveProps(newProps: ConnectionsListProps) {
+        this.updateScores(newProps);
+    }
+
+    private updateScores = (props: ConnectionsListProps) => {
+        if (props.propertySuggestionCall && (props.filterKey || props.sortMode === 'smart')) {
+            const {id, data, lang, filterKey} = props;
+            const key = filterKey.trim();
+            const properties = data.links.map(l => l.id);
+            props.propertySuggestionCall({id, key, properties, lang}).then(scores =>
+                this.setState({scores})
+            );
+        }
+    }
+
+    private get isSmartMode(): boolean {
```

Comment:
Prefer to use a simple function getter instead of a property. (Currently there are a bunch of issues with properties, like inability to override them in derived class, so you should be careful.)